### PR TITLE
AAD B2C generates the user principal to obtain the claims in the access token.

### PR DESCRIPTION
**Pull request by ZhuXiaoBing-cn**
Add the class AADB2COAuth2AuthenticatedPrincipal to store related claims carried by the access token.
From it we can get:
- tokenValue
- headers
- authorities
- attributes

### CODE REVIEW

**Comment by ZhuXiaoBing-cn** [comment]
This PR corresponds to this issue(#20007).

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-b2c-resource-server/src/main/java/com/azure/spring/sample/aad/b2c/security/WebSecurityConfiguration.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 package com.azure.spring.sample.aad.b2c.security;
 
-import com.azure.spring.aad.webapi.AADJwtBearerTokenAuthenticationConverter;
+import com.azure.spring.autoconfigure.b2c.AADB2CJwtBearerTokenAuthenticationConverter;
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-active-directory-b2c-resource-server/src/main/java/com/azure/spring/sample/aad/b2c/security/WebSecurityConfiguration.java:6]: Is the class be moved to another package?

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/b2c/AADB2COAuth2AuthenticatedPrincipal.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import java.util.Collections;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import static org.springframework.security.core.authority.AuthorityUtils.NO_AUTHORITIES;
+
+/**
+ * entity class of AADB2COAuth2AuthenticatedPrincipal
+ */
+public class AADB2COAuth2AuthenticatedPrincipal implements OAuth2AuthenticatedPrincipal, Serializable {
```
[the remark this answers is not in the document]
Comment (ZhuXiaoBing-cn) [reply sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/b2c/AADB2COAuth2AuthenticatedPrincipal.java:22]: After `JwtDecoder`, the headers, claims and authorities in JWT will be extracted and packaged into principal.

**Review by saragluna — APPROVED** [review]
LGTM.

**Comment by saragluna** [comment]
@ZhuXiaoBing-cn please also add some tests.

**Comment by saragluna** [comment]
@ZhuXiaoBing-cn, one more thing I want to make sure here is, this implementation is only for the resource server cases, right? The web application will still generate the user principal from ID token, am I understanding this correctly?

**Comment by ZhuXiaoBing-cn** [comment]
@saragluna Yes, your understanding is correct.

**Pull request merged**